File: apps/clients/android-app/app/src/main/java/com/streamverse/mobile/models/Content.kt
package com.streamverse.mobile.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * Content model representing videos (movies, TV shows, live events).
 */
@Parcelize
data class Content(
    val id: String,
    val title: String,
    val description: String,
    val genre: String,
    val category: String, // "movie", "show", "live"
    val posterUrl: String,
    val backdropUrl: String,
    val streamUrl: String, // HLS/DASH manifest URL
    val duration: Long, // in milliseconds
    val releaseYear: Int,
    val rating: Float,
    val isDrmProtected: Boolean = false,
    val drmType: String? = null, // "widevine", "fairplay", etc.
    val thumbnailUrl: String? = null,
    val cast: List<String> = emptyList(),
    val directors: List<String> = emptyList(),
    val tags: List<String> = emptyList()
) : Parcelable

/**
 * Content row for home screen categories.
 */
data class ContentRow(
    val id: String,
    val title: String,
    val items: List<Content>
)

File: apps/clients/ios-app/StreamVerse/Models/Content.swift
//
//  Content.swift
//  StreamVerse
//
//  Created on 2025-01-XX.
//

import Foundation

/// Content model representing videos (movies, TV shows, live events).
struct Content: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let description: String
    let genre: String
    let category: String // "movie", "show", "live"
    let posterUrl: String
    let backdropUrl: String
    let streamUrl: String // HLS/DASH manifest URL
    let duration: Int64 // in milliseconds
    let releaseYear: Int
    let rating: Float
    let isDrmProtected: Bool
    let drmType: String? // "fairplay", "widevine", etc.
    let thumbnailUrl: String?
    let cast: [String]
    let directors: [String]
    let tags: [String]
    
    enum CodingKeys: String, CodingKey {
        case id
        case title
        case description
        case genre
        case category
        case posterUrl
        case backdropUrl
        case streamUrl
        case duration
        case releaseYear
        case rating
        case isDrmProtected
        case drmType
        case thumbnailUrl
        case cast
        case directors
        case tags
    }
}

/// Content row for home screen categories.
struct ContentRow: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let items: [Content]
}

